src/backend/distributed/utils/colocation_utils.c

```diff
@@ -989,6 +990,70 @@ ColocatedShardIntervalList(ShardInterval *shardInterval)
 }
 
 
+/*
+ * ColocatedNonPartitionShardIntervalList function returns list of shard intervals
+ * which are co-located with given shard, except partitions. If given shard is belong
+ * to append or range distributed table, co-location is not valid for that shard.
+ * Therefore such shard is only co-located with itself.
+ */
+List *
+ColocatedNonPartitionShardIntervalList(ShardInterval *shardInterval)
+{
+	Oid distributedTableId = shardInterval->relationId;
+	List *colocatedShardList = NIL;
+
+	CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(distributedTableId);
+
+	/*
+	 * If distribution type of the table is append or range, each shard of
+	 * the shard is only co-located with itself.
+	 */
+	if (IsCitusTableTypeCacheEntry(cacheEntry, APPEND_DISTRIBUTED) ||
```

Comment:
can we add test to here as the code-cov shows we don't cover?

Reply:
This is also not coverable:
`ERROR:  distributing partitioned tables in only supported for hash-distributed tables`